routing/cross_mwm_ids.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "routing_common/transit_types.hpp"
+
+#include "base/newtype.hpp"
+#include "base/visitor.hpp"
+
+#include <cstdint>
+
+namespace routing
+{
+namespace connector
+{
+// Identifier to find a border edge in neighbouring mwm while cross mwm transition
+// for pedestrian, bicycle and car routing.
+NEWTYPE(uint64_t, OsmId);
+NEWTYPE_SIMPLE_OUTPUT(OsmId);
+
+// Identifier to find a border edge in neighbouring mwm while cross mwm transition
+// for transit routing.
+struct TransitId
+{
+  DECLARE_VISITOR_AND_DEBUG_PRINT(TransitId, visitor(m_stop1Id, "stop1_id"),
+                                  visitor(m_stop2Id, "stop2_id"), visitor(m_lineId, "line_id"))
+
+  TransitId() = default;
+  TransitId(transit::StopId stop1Id, transit::StopId stop2Id, transit::LineId lineId)
+    : m_stop1Id(stop1Id), m_stop2Id(stop2Id), m_lineId(lineId)
+  {
+  }
+
+  bool operator==(TransitId const & rhs) const
+  {
+    return m_stop1Id == rhs.m_stop1Id && m_stop2Id == rhs.m_stop2Id && m_lineId == rhs.m_lineId;
+  }
+
+  bool operator!=(TransitId const & rhs) const { return !(rhs == *this); }
+
+  bool operator<(TransitId const & rhs) const
+  {
+    if (m_stop1Id != rhs.m_stop1Id)
+      return m_stop1Id < rhs.m_stop1Id;
+    if (m_stop2Id != rhs.m_stop2Id)
+      return m_stop2Id < rhs.m_stop2Id;
+    return m_lineId < rhs.m_lineId;
+  }
+
+  bool operator>(TransitId const & rhs) const { return rhs < *this; }
+  bool operator<=(TransitId const & rhs) const { return !(rhs < *this); }
+  bool operator>=(TransitId const & rhs) const { return !(*this < rhs); }
+
+  transit::StopId m_stop1Id = transit::kInvalidStopId;
+  transit::StopId m_stop2Id = transit::kInvalidStopId;
+  transit::LineId m_lineId = transit::kInvalidLineId;
+};
+
+struct HashKey
+{
+  size_t operator()(OsmId const & key) const
+  {
+    return std::hash<OsmId::RepType>()(key.Get());
```

Comment:
`#include <functional>`

Reply:
Done

Reply:
Actually you may do following:
```c++
struct HashKey : private std::hash<OsmId::RepType>
{
  size_t operator()(OsmId const & key) const
  {
     return std::hash<...>::operator()(key);
  }
};
```
By the way, why not to reuse OsmId from generator?

Reply:
Ah, see now - we need two versions. Okay.